nixos/lib/systemd-lib.nix

```diff
@@ -335,18 +335,27 @@ in rec {
       ''}
     ''; # */
 
-  makeJobScript = name: text:
+  makeJobScript = { name, text, enableStrictShellChecks }:
     let
       scriptName = replaceStrings [ "\\" "@" ] [ "-" "_" ] (shellEscape name);
-      out = (pkgs.writeShellScriptBin scriptName ''
-        set -e
-        ${text}
-      '').overrideAttrs (_: {
+      out = (
+        if ! enableStrictShellChecks then
+          pkgs.writeShellScriptBin scriptName ''
+            set -e
```

Comment:
Naive question: Should set set -eou pipefail?

Reply:
Yeah but that can be done in another PR then. It can also cause breakage, especially `-u`.

Reply:
Yeah, that's true